src/Sarif.Converters/HdfConverter.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                string ruleId = execJsonControl.Id;
+                if (!ruleIds.Contains(ruleId))
+                {
+                    (ReportingDescriptor, IList<Result>) ruleAndResult = SarifRuleAndResultFromHdfControl(execJsonControl);
+                    rules.Add(ruleAndResult.Item1);
+                    ruleIds.Add(ruleId);
+                    results.AddRange(ruleAndResult.Item2);
+                }
+            }
+
+            return (rules, results);
+        }
+
+        private static (ReportingDescriptor, IList<Result>) SarifRuleAndResultFromHdfControl(ExecJsonControl execJsonControl)
+        {
+            var reportingDescriptor = new ReportingDescriptor
+            {
+                Id = execJsonControl.Id,
+                Name = execJsonControl.Title,
+                ShortDescription = new MultiformatMessageString
+                {
+                    Text = AppendPeriod(execJsonControl.Desc),
+                },
+                DefaultConfiguration = new ReportingConfiguration
+                {
+                    Level = SarifLevelFromHdfImpact(execJsonControl.Impact),
+                },
+                HelpUri = null,
+                Relationships = new List<ReportingDescriptorRelationship>(
+                    ((JArray)execJsonControl.Tags["nist"])
+                    .Select(p => p.ToString().Trim()).Where(p => !string.IsNullOrWhiteSpace(p)).ToList().OrderBy(o => o)
+                    .Select(s => new ReportingDescriptorRelationship()
+                    {
+                        Target = new ReportingDescriptorReference()
+                        {
+                            Id = s,
+                            ToolComponent = new ToolComponentReference()
+                            {
+                                Name = "NIST",
+                                Guid = "AAFBAB93-5201-419E-8443-D4925C542398"
+                            }
+                        },
+                        Kinds = new List<string>() { "relevant" },
+                    }))
+            };
+
+            var results = new List<Result>();
+            foreach (ControlResult controlResult in execJsonControl.Results)
+            {
+                var result = new Result
+                {
+                    RuleId = execJsonControl.Id,
+                    Message = new Message
+                    {
+                        Text = AppendPeriod(controlResult.CodeDesc),
+                    },
+                    Level = SarifLevelFromHdfImpact(execJsonControl.Impact),
+                    Rank = SarifRankFromHdfImpact(execJsonControl.Impact),
+                };
+                results.Add(result);
+            }
+            return (reportingDescriptor, results);
+        }
+
+        private static FailureLevel SarifLevelFromHdfImpact(double impact)
+        {
+            // level 4 & 5
+            // Hdf: Critical, High
+            if (impact >= 0.7)
```

Comment:
>impact >= 0.7 [](http://example.com/codeflow?start=16&length=13)

check what version of c# we are using. i think we can use the  switch with conditions

Reply:
below version I like need C# 9, your thoughts?

return impact switch
            {
                >= 0.7 => FailureLevel.Error,
                >= 0.3 => FailureLevel.Warning,
                _ => FailureLevel.Note
            };

Error	CS8400	Feature 'relational pattern' is not available in C# 8.0. Please use language version 9.0 or greater.	Sarif.Converters (net472), Sarif.Converters (netstandard2.0)	C:\GH\s-s\src\Sarif.Converters\HdfConverter.cs	163

Reply:
keep the current way :)